pkg/plugin/registry/hub.go

```diff
@@ -41,20 +45,26 @@ type Hub struct {
 	url string
 	// map of downloaded providers
 	providers map[string]ProviderDetails
-}
 
-const (
-	providerDisplayMsg = "cq-provider-%s@%s"
-)
+	// getLatestRelease is a function that will return latest release from Github repo given organization and repo name.
+	getLatestRelease LatestReleaseGetter
+}
 
 type Option func(h *Hub)
 
+func WithLatestReleaseGetter(g LatestReleaseGetter) Option {
+	return func(h *Hub) {
+		h.getLatestRelease = g
+	}
+}
+
 func NewRegistryHub(url string, opts ...Option) *Hub {
 	h := &Hub{
-		PluginDirectory: filepath.Join(".", ".cq", "providers"),
-		Logger:          logging.NewZHcLog(&zerolog.Logger, ""),
-		url:             url,
-		providers:       make(map[string]ProviderDetails),
+		PluginDirectory:  filepath.Join(".", ".cq", "providers"),
```

Comment:
This needs an update now that #456 is merged?

Reply:
Right, I'll do a rebase

Reply:
Ehm, looks like at least for this PR purposes there is no change.